common/services/event.js

```diff
@@ -0,0 +1,85 @@
+const restClient = require('../lib/api-client/rest-client')
+
+const BaseService = require('./base')
+class EventService extends BaseService {
+  postEvents(lockoutEvents, move, user) {
+    const PER_EVENTS = [
+      'PerViolentDangerous',
+      'PerWeapons',
+      'PerConcealed',
+      'PerSelfHarm',
+      'PerEscape',
+      'PerMedicalAid',
+      'PerMedicalDrugsAlcohol',
+      'PerMedicalMentalHealth',
+      'PerPropertyChange',
+      'PerGeneric',
+    ]
+    const PERSON_MOVE_EVENTS = [
+      'PersonMoveUsedForce',
+      'PersonMoveDeathInCustody',
+    ]
+
+    const todaysDate = new Date()
+    const events = []
+
+    events.push(lockoutEvents.events)
+
+    events.flat().map(async event => {
```

Comment:
Do we need to `await` this to get the results? I'm wondering what happens if one of these `POST`s fails, does it stop immediately or does it continue to the next one, and what does the user see?

Reply:
Yeah I think this should be `return Promise.all(events.flat().map(async event => {...`

Reply:
Yeah agreed adding `Promise.all` will help if any of the events sent to the api errors out. 

There is a separate ticket to deal with errors on the police custody form here [P4-3515](https://dsdmoj.atlassian.net/browse/P4-3515). We can create some nice error messaging if the submission to the API fails, telling the user to retry. We will need Sophie and Lizzie to potentially create some error messaging for this type of error as I can't see any designs for a failed submission.